### ISSUE TITLE
class.show_methods option fails in master

### ISSUE THREAD

**Issue by dur-randir** [garu/Data-Printer#124]
The following used to work

```
perl -e 'use DDP {class => {show_methods => 0}}; p $z'
```

but now it fails with q![Data::Printer] invalid value '0' for option 'show_methods'(must be one of: none,all,private,public)! message.

**Comment by garu** [comment]
@dur-randir believe it or not, this is actually a bugfix :)

The available options of `show_methods` have always been (as per the previous documentation) either *none*, *all*, *private* or *public*. Problem is it was poorly tested and allowed for any scalar to be passed. Maybe it was ok to mean *none* when *0* was passed, but what to do when *1* was passed? Or "0 but true"? or dualvar(0, "all")? The new behavior stringifies the value and only accepts the documented options.

In other words, just update `show_methods => 0` to `show_methods => 'none'` and you'll be fine :)

If you think this may require a lot of changes in a lot of code by a lot of people, let me know. We may revert the behavior to allow "0" as an alias for "none". But I'm still not sure we should :D

**Comment by dur-randir** [comment]
Probably not for many. Now I remember that I have '0' there left from my original patch that had only 0/1 as an option)
